lib/std/math.zig

```diff
@@ -1279,6 +1279,33 @@ test "lossyCast" {
     try testing.expect(lossyCast(u32, @as(f32, maxInt(u32))) == maxInt(u32));
 }
 
+/// Performs linear interpolation between *a* and *b* based on *t*.
+/// *t* must be in range 0.0 to 1.0. T must be a float type.
+///
+/// This does not guarantee returning *b* if *t* is 1 due to floating-point errors.
+/// This is monotonic.
+pub fn lerp(comptime T: type, a: T, b: T, t: T) T {
+    if (@typeInfo(T) != .Float and @typeInfo(T) != .ComptimeFloat)
```

Comment:
Could use `std.meta.isFloat` here

Reply:
That doesn't exist. Do you suggest adding it? Because I think it's a great idea. Same for integers.

Reply:
Oops, yeah, sorry, typo